### PR TITLE
Match several functions in code_80005FD0

### PR DESCRIPTION
**Pull request by Tmcg2**
Intermediate commit for matching a ton of waypoint (checkpoint?) related changes
 This commit should be fixup'd/squashed prior to merging

Signed-off-by: Taggerung <tyler.taggerung@gmail.com>

### CODE REVIEW

**Comment by Tmcg2** [comment]
This PR is ready now, I don't intend to add more changes to it

File: include/waypoints.h [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+struct TrackWayPoint {
+    /* 0x00 */ s16 wayPointX;
+    /* 0x02 */ s16 wayPointY;
+    /* 0x04 */ s16 wayPointZ;
+    /* 0x06 */ u16 wayPointTrackSegment;
+}; // size = 0x08
+
+/**
+ * These are per-path arrays that contain some information relating to waypoints
```
[the remark this answers is not in the document]
Comment (MegaMech) [reply include/waypoints.h:12]: https://github.com/RenaKunisaki/mariokart64/wiki/Course-Paths

File: include/waypoints.h [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+// So, in 2 Player Grand Prix, the first 2 entries are always 0
+extern u16 D_80163318[];
+// Seems to be a per-path overcount of the waypoint count
+extern s32 D_80163368[];
+// Seemingly the Z position of the 1st waypoint in the 0th path?
+extern f32 D_8016344C;
+// These seem to track whether a player has entered or exited the "unknown" zone in yoshi's valley
+// See func_80009000 and func_800090F0
+// Is 1 when a player is in the "unknown" zone in yoshi's valley
+extern s16 D_80163490[];
```
[the remark this answers is not in the document]
Comment (Tmcg2) [reply include/waypoints.h:83]: The former. As you approach the first split in the track you enter the "unknown zone" (I'm sure there's a better name for it)
Shortly after you pass the giant egg you exit the zone.

File: include/waypoints.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tmcg2) [reply include/waypoints.h:12]: As best as I can tell, `D_80164550` is the "true" location of the waypoints for all 4 possible paths. `D_80164560` and `D_80164570` also have entries for all 4 paths, but its not clear to me what those entries actually track.
`func_80010218` appears responsible for setting the values of `D_80164560` and `D_80164570`, but I don't understand the math going on.

File: include/waypoints.h [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+// Tracks something on a per-player basis, no idea what though
+extern f32 D_80163068[];
+// Track segment by playerId, although it curiously does NOT track values for human players
+// So, in 2 Player Grand Prix, the first 2 entries are always 0
+extern u16 D_80163318[];
+// Seems to be a per-path overcount of the waypoint count
+extern s32 D_80163368[];
+// Seemingly the Z position of the 1st waypoint in the 0th path?
+extern f32 D_8016344C;
```
[the remark this answers is not in the document]
Comment (Tmcg2) [reply include/waypoints.h:79]: Maybe? I can see where its given a value and what that value hence (hence the comment) but I  can't make head nor tail of its usage.

File: include/waypoints.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (MegaMech) [reply include/waypoints.h:83]: "Maze" zone perhaps. I think these comments are good enough for now. Someone can always come back over it later and clarify them further.

**Review by MegaMech — APPROVED** [review]

**Pull request merged**